components/generators/argocd/__init__.py

```diff
@@ -15,7 +15,7 @@ def j2(filename, ctx):
     return render_jinja2_file(filename, ctx, search_paths=search_paths)
 
 
-inv = inventory()
+inv = inventory(lazy=True)
```

Comment:
ideally, we'd want to not use lazy

Reply:
I intend to refactor it heavily to move to the new BaseClass, as part of that we might achieve this as well

Reply:
got it